src/commands/config.rs

```diff
@@ -0,0 +1,161 @@
+//!     "farm_directory": "/Users/abhi3700/Library/Application Support/pulsar/farms",
+//!     "farm_size": "3.0 GB"
+//!   },
+//!   "node": {
+//!     "directory": "/Users/abhi3700/Library/Application Support/pulsar/node",
+//!     "name": "abhi3700"
+//!   }
+//! }
+//! in file: "/Users/abhi3700/Library/Application Support/pulsar/settings.toml"
+//! ```
+//!
+//! ### Chain
+//! ```
+//! $ pulsar config -c devnet
+//! ```
+//!
+//! ### Farm size
+//! ```
+//! $ pulsar config -f 3GB
+//! ```
+//!
+//! ### Reward address
+//!
+//! ```
+//! $ pulsar config -r 5CDstQSbxPrPAaRTuVR2n9PHuhGYnnQvXdbJSQmodD5i16x2
+//! ```
+//!
+//! ### Node directory
+//! ```
+//! $ pulsar config -n "/Users/abhi3700/test/pulsar1/node"
+//! ```
+//!
+//! ### Farm directory
+//! ```
+//! $ pulsar config -d "/Users/abhi3700/test/pulsar1/farms"
+//! ```
+//!
+//! ### All params
+//! ```
+//! $ pulsar config \
+//!   --chain devnet \
+//!   --farm-size 5GB \
+//!   --reward-address 5DXRtoHJePQBEk44onMy5yG4T8CjpPaK4qKNmrwpxqxZALGY \
+//!   --node-directory "/Users/abhi3700/test/pulsar1/node" \
+//!   --farm-directory "/Users/abhi3700/test/pulsar1/farms"
+//! ```
+
+use std::fs;
+use std::path::PathBuf;
+use std::str::FromStr;
+
+use color_eyre::eyre::{self, bail};
+
+use crate::commands::wipe::wipe;
+use crate::config::{parse_config, parse_config_path, ChainConfig, Config};
+use crate::utils::{create_or_move_data, reward_address_parser, size_parser};
+
+// function for config cli command
+pub(crate) async fn config(
+    show: bool,
+    chain: Option<String>,
+    farm_size: Option<String>,
+    reward_address: Option<String>,
+    node_dir: Option<String>,
+    farm_dir: Option<String>,
+) -> eyre::Result<()> {
+    // Define the path to your settings.toml file
+    let config_path = parse_config_path()?;
+
+    // if config file doesn't exist, then throw error
+    if !config_path.exists() {
+        bail!("Config file: \"settings.toml\" not found.\nPlease use `pulsar init` command first.");
+    }
+
+    // Load the current configuration
+    let mut config: Config = parse_config()?;
+
+    if show {
+        // Display the current configuration as JSON
+        // Serialize `config` to a pretty-printed JSON string
+        let serialized = serde_json::to_string_pretty(&config)?;
+        println!(
+            "Current Config set as: \n{}\nin file: {:?}",
+            serialized,
+            config_path.to_str().expect("Expected stringified config path")
+        );
+    } else {
+        // no options provided
+        if chain.is_none()
+            && farm_size.is_none()
+            && reward_address.is_none()
+            && node_dir.is_none()
+            && farm_dir.is_none()
+        {
+            println!("At least one option has to be provided.\nTry `pulsar config -h`");
+            return Ok(());
+        }
+
+        // update (optional) the chain
+        if let Some(c) = chain {
+            config.chain = ChainConfig::from_str(&c)?;
```

Comment:
Maybe check if the chain has changed before wiping it?

Reply:
Although redundant.
[Commit](https://github.com/subspace/pulsar/commit/ef9072f468a1496d026ba1705289f26ca1533497).